python/cugraph/graph/c_graph.pyx

```diff
@@ -94,6 +113,41 @@ class Graph:
         self.delete_transposed_adj_list()
         free(g)
 
+    def renumber(self, source_col, dest_col):
+        cdef gdf_column * src_renumbered = NULL
+        cdef gdf_column * dst_renumbered = NULL
+        cdef gdf_column * numbering_map = NULL
+
+        cdef gdf_column source = get_gdf_column_view(source_col)
+        cdef gdf_column dest = get_gdf_column_view(dest_col)
+
+        """
+        cdef uintptr_t source = create_column(source_col)
+        cdef uintptr_t dest = create_column(dest_col)
+
+        err = gdf_renumber_vertices(< const gdf_column * > source,
+	      			    < const gdf_column * > dest,
+        """
+        err = gdf_renumber_vertices(&source,
+	      			    &dest,
+				    &src_renumbered,
+				    &dst_renumbered,
+				    &numbering_map)
+
+        cudf.bindings.cudf_cpp.check_gdf_error(err) 
+
+        src_renumbered_array = rmm.device_array_from_ptr(<uintptr_t> src_renumbered.data,
+                                     nelem=src_renumbered.size,
+                                     dtype=dtypes_inv[src_renumbered.dtype])
+        dst_renumbered_array = rmm.device_array_from_ptr(<uintptr_t> dst_renumbered.data,
+                                     nelem=dst_renumbered.size,
+                                     dtype=dtypes_inv[dst_renumbered.dtype])
+        numbering_map_array = rmm.device_array_from_ptr(<uintptr_t> numbering_map.data,
+                                     nelem=numbering_map.size,
+                                     dtype=dtypes_inv[numbering_map.dtype])
+
```

Comment:
So this function is responsible for freeing src_renumbered, dst_renumbered, and numbering_map allocated by gdf_renumber_vertices, but fail to do that. As I mentioned in the implementation of gdf_renumber_vertices, a better way is to avoid memory allocations.

Don't get confused between allocating memory for the gdf_column structure and allocating actual column array. The gdf_column structure has fixed size and is not really that large, so no need to worry about blowing out stack. Also, src_renumbered.data, dst_renumbered.data, and numbering_map.data should not be freed in this function but src_renumbered, dst_renumbered, and numbering_map are used only temporarily inside this function.

Reply:
Changed gdf_column to be local, so that should (I think) take care of that.  Should be in next push.